Week3/homework/router.js

```diff
@@ -0,0 +1,185 @@
+'use strict';
+
+const Express = require('express');
+const router = Express.Router();
+const validator = require('./validator');
+const { validationResult } = require('express-validator/check');
+const {
+  userList,
+  complete,
+  deleteItem,
+  getItems,
+  createItem,
+  deleteList,
+  validateListID,
+  createList,
+  userID,
+  insertUser,
+} = require('./query');
+
+router
+  .get('/', async (request, response) => {
+    try {
+      if (request.session.username) {
+        const uid = await userID(request.session.username);
+        request.session.items = await getItems(uid);
+        const { message, items } = request.session;
+        return response.status(200).json({ message, items });
+      }
+      if (request.session.message) {
+        const { message } = request.session;
+        return response.status(400).json({ message });
+      }
+      response.status(200).json('Please sign in');
+    } catch (error) {
+      console.log(error.message);
+    }
+  })
+
+  .post('/register', validator.registration, async (request, response) => {
+    try {
+      const { name, username, password } = await request.body;
+      const isExists = await userList(username, password);
+      const errors = validationResult(request);
+      if (!errors.isEmpty()) {
+        return response.status(400).json({ error: errors.array() });
+      }
+      if (isExists.userCheck !== username && errors.isEmpty()) {
+        await insertUser(name, username, password);
+        request.session.message = 'user registered';
+        request.session.username = username;
+        return response.redirect('/');
+      }
+      request.session.message = 'username already registered';
+      return response.redirect('/');
+    } catch (error) {
+      console.log(error.message);
+    }
+  })
+
+  .post('/signin', validator.login, async (request, response) => {
+    try {
+      const { username, password } = await request.body;
+      const users = await userList(username, password);
+
+      const errors = validationResult(request);
+      if (!errors.isEmpty()) {
+        return response.status(400).json({ error: errors.array() });
+      }
+      if (users.userCheck === username && users.passCheck === password && errors.isEmpty()) {
+        request.session.username = users.userCheck;
+        request.session.message = `Welcome ${users.name}`;
+        return response.redirect('/');
+      }
+      response.json('Please sign in / sign up');
+    } catch (error) {
+      console.log(error.message);
+    }
+  })
+
+  .get('/signout', async (request, response) => {
```

Comment:
I would personally also use POST for logout too but amazing idea to provide these login and logout functionality :)